### PR TITLE
SELC-3062 Add createInstitution from ANAC Api

### PR DESCRIPTION
**Pull request by flaminiaScarciofolo**
#### List of Changes

Add api for create SA institution from ANAC

### CODE REVIEW

**Review by gianmarcoplutino — DISMISSED** [review]

File: web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java [hunk 1]
```diff
@@ -124,6 +124,28 @@ public ResponseEntity<InstitutionResponse> createInstitutionFromIpa( @RequestBod
...
+     * @return InstitutionResponse
+     * * Code: 201, Message: successful operation, DataType: InstitutionResponse
+     * * Code: 404, Message: Institution data not found on Ipa, DataType: Problem
+     * * Code: 400, Message: Bad Request, DataType: Problem
+     * * Code: 409, Message: Institution conflict, DataType: Problem
+     */
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiOperation(value = "${swagger.mscore.institution.create.from-anac}", notes = "${swagger.mscore.institution.create.from-anac}")
+    @PostMapping(value = "/from-anac/")
+    public ResponseEntity<InstitutionResponse> createInstitutionFromAnac(@RequestBody @Valid InstitutionRequest institution) {
```
Comment (manuraf) [inline web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java:140]: Why we need an InstitutionRequest if we use data from SaResource for saving institution on createInstitutionFromAnac?

File: web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java [hunk 2]
```diff
@@ -124,6 +124,28 @@ public ResponseEntity<InstitutionResponse> createInstitutionFromIpa( @RequestBod
...
+        CustomExceptionMessage.setCustomMessage(GenericError.CREATE_INSTITUTION_ERROR);
+        if (!StringUtils.hasText(institution.getTaxCode())) {
+            throw new ValidationException("TaxCode is required");
+        }
```
Comment (manuraf) [inline web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java:144]: you might add @NotEmpty at InstitutionRequest

File: core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/CreateInstitutionStrategyAnac.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+import static it.pagopa.selfcare.mscore.constant.GenericError.CREATE_INSTITUTION_ERROR;
+
+@Slf4j
+public class CreateInstitutionStrategyAnac implements CreateInstitutionStrategy {
+    private final PartyRegistryProxyConnector partyRegistryProxyConnector;
+
+    private final InstitutionConnector institutionConnector;
+
+    private Institution institution;
```
Comment (manuraf) [inline core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/CreateInstitutionStrategyAnac.java:25]: you might remove institution object if it is not used

File: core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/factory/CreateInstitutionStrategyFactory.java [hunk 4]
```diff
@@ -35,5 +36,10 @@ public CreateInstitutionStrategy createInstitutionStrategy(Institution instituti
     public CreateInstitutionStrategy createInstitutionStrategyIpa() {
         return new CreateInstitutionStrategyIpa(partyRegistryProxyConnector, institutionConnector, institutionMapper);
     }
-  
+
+    public CreateInstitutionStrategy createInstitutionStrategyAnac(Institution institution) {
+        CreateInstitutionStrategyAnac strategy = new CreateInstitutionStrategyAnac(partyRegistryProxyConnector, institutionConnector);
+        strategy.setInstitution(institution);
```
Comment (manuraf) [inline core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/factory/CreateInstitutionStrategyFactory.java:42]: ditto

**Review by manuraf — CHANGES_REQUESTED** [review]

File: core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/CreateInstitutionStrategyAnac.java (hunk 3 above)
Comment (flaminiaScarciofolo) [reply core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/CreateInstitutionStrategyAnac.java:25]: Why isn't it used?

Comment (manuraf) [reply core/src/main/java/it/pagopa/selfcare/mscore/core/strategy/CreateInstitutionStrategyAnac.java:25]: at row 44 you set institution as a new object returned by addFieldsToInstitution.....you don't use the object passed in constructor

File: web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java (hunk 2 above)
Comment (manuraf) [reply web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java:144]: you may also introduce this annotation for other api.....taxCode must always be valued

File: web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java (hunk 1 above)
Comment (manuraf) [reply web/src/main/java/it/pagopa/selfcare/mscore/web/controller/InstitutionController.java:140]: i have looked at CreateInstitutionStrategyAnac and institution fields aren't used....try to check there

File: web/src/main/java/it/pagopa/selfcare/mscore/web/model/institution/InstitutionRequest.java [hunk 5]
```diff
@@ -19,7 +20,10 @@ public class InstitutionRequest {
     private String digitalAddress;
     private String address;
     private String zipCode;
+
+    @NotEmpty
```
Comment (manuraf) [inline web/src/main/java/it/pagopa/selfcare/mscore/web/model/institution/InstitutionRequest.java:24]: you might add an error message

**Review by manuraf — CHANGES_REQUESTED** [review]

**Review by manuraf — APPROVED** [review]

**Pull request merged**